### PR TITLE
Add sample error mapping

### PR DESCRIPTION
**Pull request by islean**
## Description

We are not mapping errors found in standalone sample orders properly.

### Added

-

### Changed

- Sample errors are mapped to their field in the response

### Fixed

-

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/services/order_validation_service/response_mapper.py [hunk 1]
```diff
@@ -74,8 +93,8 @@ def wrap_case_fields(case: dict) -> None:
             set_field(entity=case, field=field, value=value)
 
 
-def wrap_sample_fields(case: dict) -> None:
-    samples: list[dict] = case.get("samples", [])
+def wrap_sample_fields(dict_containing_samples: dict) -> None:
+    samples: list[dict] = dict_containing_samples.get("samples", [])
```
Comment (islean) [inline cg/services/order_validation_service/response_mapper.py:97]: This got quite ugly but essentially the only input needed is a dict with a key called "samples". Any way to make it better?

File: cg/services/order_validation_service/models/sample.py [hunk 2]
```diff
@@ -9,7 +9,6 @@ class Sample(BaseModel):
     container: ContainerEnum
     container_name: str | None = None
     name: str = Field(pattern=NAME_PATTERN, min_length=2, max_length=128)
-    require_qc_ok: bool
```
Comment (islean) [inline cg/services/order_validation_service/models/sample.py:12]: Turns out that this is not a setting for all workflows

File: cg/services/order_validation_service/response_mapper.py [hunk 3]
```diff
@@ -47,13 +59,20 @@ def get_case(order: dict, index: int) -> dict:
     return order["cases"][index]
 
 
-def get_sample(case: dict, index: int) -> dict:
+def get_case_sample(case: dict, index: int) -> dict:
     return case["samples"][index]
 
 
+def get_sample(order: dict, index: int) -> dict:
+    return order["samples"][index]
```
[the remark this answers is not in the document]
Comment (Vince-janv) [reply cg/services/order_validation_service/response_mapper.py:67]: Definitely separate. I can't imagine a naming that would make it readable as one function

File: cg/services/order_validation_service/response_mapper.py (hunk 1 above)
Comment (Vince-janv) [reply cg/services/order_validation_service/response_mapper.py:97]: Does the input need to be a dict? Can we not just pass the samples value if it exists? Or was there a decision to only operate on the whole order?

**Review by Vince-janv — APPROVED** [review]
Looks good 👍

File: cg/services/order_validation_service/response_mapper.py (hunk 1 above)
Comment (islean) [reply cg/services/order_validation_service/response_mapper.py:97]: ooo that sounds promising! I will give it a shot

**Comment by islean** [comment]
Seems to map the errors correctly:

image

File: cg/services/order_validation_service/response_mapper.py (hunk 1 above)
Comment (islean) [reply cg/services/order_validation_service/response_mapper.py]: Change worked!

**Pull request merged**